ipaserver/plugins/cert.py

```diff
@@ -860,6 +866,33 @@ def execute(self, csr, all=False, raw=False, chain=False, **kw):
         )
 
 
+def _emails_are_valid(subject_emails, san_emails, principal_emails):
+    """
+    Checks if any email addr from DN or SAN extension does not
+    appear in ldap entry, comparing the domain part case-insensitively.
+    """
+
+    if not any(principal_emails):
+        return False
+
+    def lower_domain(email):
+        if '@' not in email:
+            return ''
+
+        email_splited = email.split('@')
+        return '{}@{}'.format(email_splited[0], email_splited[1].lower())
```

Comment:
I did just a little bit different because the example above it doesn't work. The `split` method would return a list like `["fbarreto", "redhat.com"]`, so, it's necessary to join the two parts with a `@`.